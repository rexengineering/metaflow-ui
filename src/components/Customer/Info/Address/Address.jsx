import { Typography } from "@material-ui/core";
import React from "react";
import PropTypes from "prop-types";
import InfoSection from "../InfoSection";

function Address({ address, className }) {
  return (
    <InfoSection className={className} title="Address">
      <Typography variant="h6">{address}</Typography>
    </InfoSection>
  );
}

Address.defaultProps = {
  className: "",
};

Address.propTypes = {
  address: PropTypes.string.isRequired,
  className: PropTypes.string,
};

export default Address;
