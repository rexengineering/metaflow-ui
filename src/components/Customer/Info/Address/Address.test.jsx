import { render, screen } from "@testing-library/react";
import React from "react";
import Address from "./Address";

describe("<Address />", () => {
  const renderAddress = (address = "Mty 156") => {
    return <Address address={address} />;
  };

  it("should render", () => {
    const { asFragment } = render(renderAddress());
    const initialRender = asFragment();

    expect(initialRender).not.toBeNull();
  });

  it("should match snapshot", () => {
    const { asFragment } = render(renderAddress());
    const initialRender = asFragment();

    expect(initialRender).toMatchSnapshot();
  });

  it("should render provided Address", () => {
    const address = "MX 1528";
    render(renderAddress(address));
    const addressNode = screen.getByText(address);
    expect(addressNode).toBeInTheDocument();
  });
});
