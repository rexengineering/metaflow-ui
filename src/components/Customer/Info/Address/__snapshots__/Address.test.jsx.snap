// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Address /> should match snapshot 1`] = `
<DocumentFragment>
  <section
    class=""
  >
    <div
      class="makeStyles-header-3"
    >
      <h6
        class="MuiTypography-root makeStyles-label-4 MuiTypography-h6"
      >
        Address
      </h6>
    </div>
    <div>
      <h6
        class="MuiTypography-root MuiTypography-h6"
      >
        Mty 156
      </h6>
    </div>
  </section>
</DocumentFragment>
`;
